`General`: Add capability to remove course icon

rriyaldhi opened the pull request:
<!-- Thanks for contributing to Artemis! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->
<!-- If your pull request is not ready for review yet, create a draft pull request! -->

### Checklist
#### General
<!-- You only need to choose one of the first two check items: Generally, test on the test servers. -->
<!-- If it's only a small change, testing it locally is acceptable and you may remove the first checkmark. If you are unsure, please test on the test servers. -->
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->

[#5526](https://github.com/ls1intum/Artemis/issues/5526)

### Description
<!-- Describe your changes in detail -->

When Delete Icon Button is clicked, the course icon is set to null. When the Save Button is clicked, the Course Update API will be called. The Course Update API will set the course icon to null and delete the previous course icon file.

### Steps for Testing
<!-- Please describe in detail how the reviewer can test your changes. -->
Prerequisites:
- 1 Instructor
- 1 Course with Icon

1. Log in to Artemis
2. Navigate to Course Administration
3. Open Course Detail page
4. Click Edit Button
5. Click Delete Button under the Course Icon
6. Click Save
7. The Course Icon should be removed

### Review Progress
<!-- Each Pull Request should be reviewed by at least two other developers. The code as well as the functionality (= manual test) needs to be reviewed. -->
<!-- The reviewer or author check the following boxes depending on what was reviewed or tested. All boxes should be checked before merge. -->
<!-- You can add additional checkboxes if it makes sense to only review parts of the code or functionality. -->
<!-- When changes are pushed, uncheck the affected boxes. (Not all changes require full re-reviews.) -->

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage
<!-- Please add the test coverages for all changed files here. You can see this when executing the tests locally (see build.gradle and package.json) or when looking into the corresponding Bamboo build plan. -->
<!-- Lines are the main reference but a significantly lower branch percentage can indicate missing edge cases in the tests. -->
| Class/File | Branch | Line |
|------------|-------:|-----:|
| course-update.component.ts | 70.24% | 87% |

### Screenshots
<!-- Add screenshots to demonstrate the changes in the UI. -->
<!-- Create a GIF file from a screen recording in a docker container https://toub.es/2017/09/11/high-quality-gif-with-ffmpeg-and-docker/ -->
**Desktop**
<img width="1425" alt="Screen Shot 2022-10-19 at 2 45 12 PM" src="https://user-images.githubusercontent.com/10885503/196694701-82c7e185-a653-4bec-901c-c37d5fc4762e.png">

**Mobile**
<img width="376" alt="Screen Shot 2022-10-19 at 2 45 28 PM" src="https://user-images.githubusercontent.com/10885503/196694726-005e23f1-aee2-4bb5-ae4b-0abed088afe1.png">

ge65cer reviewed (commented):
Tested on TS1, works! My suggestions:
* Remove the confirmation dialog, as clicking on "cancel" aborts the course editing and the icon is still there. You already need to confirm deleting by clicking on "save".
* Move the button next to the upload form, maybe label it "Remove Icon". In my opinion it can also does not need to be red colored, as it is not a destructive action.
<img width="300" alt="Screenshot 2022-10-07 at 21 46 12" src="https://user-images.githubusercontent.com/102237363/194642136-cb833ba4-a0c6-488d-b4eb-068a18e0643a.png">

Feel free to wait for feedback from other reviewers before applying these requests.

JohannesStoehr commented on src/main/webapp/app/course/manage/course-update.component.ts:
```diff
@@ -482,6 +489,19 @@ export class CourseUpdateComponent implements OnInit {
+    deleteCourseIcon() {
+        this.course.courseIcon = undefined;
+        this.courseImageFile = undefined;
+        this.courseImageFileName = this.course.courseIcon!;
```
```suggestion
        this.courseImageFileName = undefined;
```

JohannesStoehr commented on src/main/webapp/app/course/manage/course-update.component.ts:
```diff
@@ -482,6 +489,19 @@ export class CourseUpdateComponent implements OnInit {
+        this.course.courseIcon = undefined;
+        this.courseImageFile = undefined;
+        this.courseImageFileName = this.course.courseIcon!;
+        this.croppedImage = '';
```
Why do you set this to an empty string instead of undefined as the rest?

rriyaldhi replied:
croppedImage is used here: [course-update.component.ts#L274](https://github.com/ls1intum/Artemis/blob/52869217a3c555e19eeda0d69607a771197e9fc9/src/main/webapp/app/course/manage/course-update.component.ts#L274), setting it to undefined will cause it to break. Also I set it to empty string, to make it consistent with the other assignment (e.g: [course-update.component.ts#L188](https://github.com/ls1intum/Artemis/blob/52869217a3c555e19eeda0d69607a771197e9fc9/src/main/webapp/app/course/manage/course-update.component.ts#L188)).

JohannesStoehr commented on src/test/cypress/integration/CourseManagement.spec.ts:
```diff
@@ -121,4 +122,56 @@ describe('Course management', () => {
+            cy.get('.no-image').should('exist');
+            cy.intercept(PUT, BASE_API + 'courses').as('updateCourseQuery');
+            cy.get('#save-entity').click();
+            cy.wait('@updateCourseQuery').then((_) => {
```
I think you can also leave out the _ completely when it’s unused

JohannesStoehr reviewed (changes requested):
Looked through the code and left some comments. 
I mostly agree with Moritz‘ comments, but I actually like the red color of the button, since you’re still deleting something, even though you have to save the modification afterwards.

ge65cer commented:
Related to https://github.com/ls1intum/Artemis/pull/5733

rriyaldhi commented:
> Tested on TS1, works! My suggestions:
> 
> * Remove the confirmation dialog, as clicking on "cancel" aborts the course editing and the icon is still there. You already need to confirm deleting by clicking on "save".
> * Move the button next to the upload form, maybe label it "Remove Icon". In my opinion it can also does not need to be red colored, as it is not a destructive action.
> 
> <img alt="Screenshot 2022-10-07 at 21 46 12" width="300" src="https://user-images.githubusercontent.com/102237363/194642136-cb833ba4-a0c6-488d-b4eb-068a18e0643a.png">
> 
> Feel free to wait for feedback from other reviewers before applying these requests.

Hi @ge65cer , I've moved the button next to the upload form, changed the label to Remove Icon, and removed the confirmation dialog. But I keep the red color as it is still a deleting action.

JohannesStoehr reviewed (approved):
Code changes look good to me.
We might think about changing the German "Kursicon" to a more German word in the future

Santia-go reviewed (dismissed):
Tested on TS1 
Working as expected!

ge65cer reviewed (approved):
Re-tested on TS1, thanks for applying my feedback.

stefanwaldhauser reviewed (approved):
Reapprove

LiaLism reviewed (approved):
Removing the icon worked. Tested on ts1.

JohannesStoehr reviewed (approved):
Rechecked thee code changes, looks good to me
